### PR TITLE
Feature: Add vault secrets api with Kaigara

### PR DESCRIPTION
**Pull request by jengjeng**

### CODE REVIEW

File: skel/handlers/index.go [hunk 1]
```diff
@@ -31,6 +32,11 @@ func Setup(app *sonic.Runtime) {
 	router.GET("/version", version)
 
 	SetPageRoutes(router)
+
+	opx := router.Group("/api/v2/vault")
```
Comment (vshatravenko) [inline skel/handlers/index.go:36]: I think `/api/v2/config` would be better
We'd need to have endpoints for public(env.js) and all scopes(tower)

Comment (vshatravenko) [inline skel/handlers/index.go:36]: Why are we using opx here?

**Review by vshatravenko — CHANGES_REQUESTED** [review]

File: skel/handlers/vault.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+		}
+	}
+
+	ctx.JSON(http.StatusOK, result)
+}
+
+func parseScopes(scopes string) []string {
+	return strings.Split(scopes, ",")
+}
```
Comment (vshatravenko) [inline skel/handlers/vault.go:72]: We don't need this since we're using all scopes

**Review by vshatravenko — CHANGES_REQUESTED** [review]

File: skel/handlers/index.go (hunk 1 above)
Comment (jengjeng) [reply skel/handlers/index.go]: I saw we move to /admin. I will leave it as it is.

File: skel/handlers/index.go [hunk 3]
```diff
@@ -31,6 +40,32 @@ func Setup(app *sonic.Runtime) {
...
+	// Define all public env on first system start
+	WriteCache(vaultService, scope, true)
+	go StartConfigCaching(vaultService, scope)
+}
+
+// StartConfigCaching will fetch latest data from vault every 30 seconds
+func StartConfigCaching(vaultService *vault.Service, scope string) {
+	for {
+		<-time.After(30 * time.Second)
+		WriteCache(vaultService, scope, false)
```
Comment (calj) [inline skel/handlers/index.go:67]: You need a sync.RWMutex
Here you do lock.Lock() and lock.Unlock()

File: skel/handlers/public.go [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+)
+
+// GetPublicConfigs returns public configs
+func GetPublicConfigs(ctx *gin.Context) {
+	ctx.JSON(http.StatusOK, memoryCache)
```
Comment (calj) [inline skel/handlers/public.go:11]: here you do a lock.Rlock() and lock.RUnlock()

**Pull request merged**